context/context.go

```diff
@@ -194,20 +195,20 @@ func NewContextWD(rt RootType) (*Context, error) {
 		return nil, ErrOldVersion{`Use the "migrate" command to update.`}
 	}
 
-	return NewContextRoot(root)
+	return NewContextRoot(base, root)
 }
 
 // NewContextRoot creates a new context for the given root folder.
-func NewContextRoot(root string) (*Context, error) {
+func NewContextRoot(base *Context, root string) (*Context, error) {
 	pathToVendorFile := filepath.Join("vendor", vendorFilename)
 	vendorFolder := "vendor"
 
-	return NewContext(root, pathToVendorFile, vendorFolder, false)
+	return NewContext(base, root, pathToVendorFile, vendorFolder, false)
 }
 
 // NewContext creates new context from a given root folder and vendor file path.
 // The vendorFolder is where vendor packages should be placed.
-func NewContext(root, vendorFilePathRel, vendorFolder string, rewriteImports bool) (*Context, error) {
+func NewContext(base *Context, root, vendorFilePathRel, vendorFolder string, rewriteImports bool) (*Context, error) {
```

Comment:
If we do the above, probably need another param here.
Or separate out NewContext into two parts, (1) init struct (2) inspect paths.